### PR TITLE
refactor(pod_cpu_hog): Converting pod_cpu_hog job to pod

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Converting pod_cpu_hog job to pod

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ispeakc0de** [comment]
```
2020-03-30 18:23:40.834178 ************ BRACE YOURSELF, EXPERIMENT BEGINS! ************ 

2020-03-30 18:23:43.668968 Step: Gathering Facts 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:2
META: ran handlers

2020-03-30 18:23:44.026332 Step: Identify the chaos util to be invoked 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_prerequisites.yml:1
details => {"changed": true, "checksum": "7902a07c332b69314ed12c9c1accd3dfed8c128e", "dest": "./chaosutil.yml", "gid": 0, "group": "root", "md5sum": "5be0ff448e1a46102e3d3f8605c51d57", "mode": "0644", "owner": "root", "size": 49, "src": "/root/.ansible/tmp/ansible-tmp-1585592623.71-223384506326761/source", "state": "file", "uid": 0}

2020-03-30 18:23:44.066341 Step: include_vars 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:27
details => {"ansible_facts": {"c_util": "/chaoslib/litmus/cpu_hog/pod_cpu_hog.yml"}, "ansible_included_var_files": ["/experiments/generic/pod_cpu_hog/chaosutil.yml"], "changed": false}

2020-03-30 18:23:44.112364 Step: Construct chaos result name 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:31
details => {"ansible_facts": {"c_result": "engine-pod-cpu-hog"}, "changed": false}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-30 18:23:44.356796 Step: [PreReq]: Generate the chaos result CR to reflect SOT (Start of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:3
details => {"changed": true, "checksum": "e0e6ad93524e954f1fdd284ca378ba5585df497f", "dest": "/utils/runtime/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "b9e74c4cc8088cef9f81d737632e9b4b", "mode": "0644", "owner": "root", "size": 306, "src": "/root/.ansible/tmp/ansible-tmp-1585592624.19-159003196477134/source", "state": "file", "uid": 0}

2020-03-30 18:23:45.376939 Step: [PreReq]: Apply the chaos result CR for pod-cpu-hog experiment 
task path: /utils/runtime/update_chaos_result_resource.yml:13
details => {"changed": true, "cmd": "kubectl apply -f /utils/runtime/chaos-result.yml -n shubham", "delta": "0:00:00.799550", "end": "2020-03-30 18:23:45.356101", "failed_when_result": false, "rc": 0, "start": "2020-03-30 18:23:44.556551", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/engine-pod-cpu-hog configured", "stdout_lines": ["chaosresult.litmuschaos.io/engine-pod-cpu-hog configured"]}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-30 18:23:46.014099 Step: [Status]: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n default -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.388448", "end": "2020-03-30 18:23:45.991756", "rc": 0, "start": "2020-03-30 18:23:45.603308", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-30 18:23:46.531101 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n default -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.375891", "end": "2020-03-30 18:23:46.511311", "rc": 0, "start": "2020-03-30 18:23:46.135420", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-30 18:23:47.003553 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.295092", "end": "2020-03-30 18:23:46.988118", "rc": 0, "start": "2020-03-30 18:23:46.693026", "stderr": "", "stderr_lines": [], "stdout": "2020-03-30T18:23:46+00:00", "stdout_lines": ["2020-03-30T18:23:46+00:00"]}

2020-03-30 18:23:47.045921 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-30T18:23:46Z"}, "changed": false}

2020-03-30 18:23:47.686971 Step: [Event]: Creating a PreChaosCheck event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-30T18:23:46Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "8d079fe8-72b3-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-30T18:23:46Z", "message": "AUT is Running successfully", "metadata": {"creationTimestamp": "2020-03-30T17:15:05Z", "name": "event-\"PreChaosCheck\"", "namespace": "shubham", "resourceVersion": "2076", "selfLink": "/api/v1/namespaces/shubham/events/event-%22PreChaosCheck%22", "uid": "00104ea1-72aa-11ea-91b9-42010a800049"}, "reason": "PreChaosCheck", "reportingComponent": "", "reportingInstance": "", "source": {"component": "pod-cpu-hog-sy5iws-k4p9t"}, "type": "Normal"}}
included: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml for 127.0.0.1

2020-03-30 18:23:48.269958 Step: Select the app pod 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:6
details => {"changed": true, "cmd": "kubectl get pod -l run=nginx -n default -o=custom-columns=:metadata.name --no-headers | shuf | head -1", "delta": "0:00:00.366875", "end": "2020-03-30 18:23:48.253491", "rc": 0, "start": "2020-03-30 18:23:47.886616", "stderr": "", "stderr_lines": [], "stdout": "nginx-6546cd7545-jsrzk", "stdout_lines": ["nginx-6546cd7545-jsrzk"]}

2020-03-30 18:23:48.319312 Step: Record app pod name 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:15
details => {"ansible_facts": {"app_pod": "nginx-6546cd7545-jsrzk"}, "changed": false}

2020-03-30 18:23:48.816123 Step: Identify the application node 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:24
details => {"changed": true, "cmd": "kubectl get pod nginx-6546cd7545-jsrzk -n default --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.364621", "end": "2020-03-30 18:23:48.798972", "rc": 0, "start": "2020-03-30 18:23:48.434351", "stderr": "", "stderr_lines": [], "stdout": "gke-shubham-test-default-pool-c91a8a24-762n", "stdout_lines": ["gke-shubham-test-default-pool-c91a8a24-762n"]}

2020-03-30 18:23:48.864515 Step: set_fact 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:32
details => {"ansible_facts": {"app_node": "gke-shubham-test-default-pool-c91a8a24-762n"}, "changed": false}

2020-03-30 18:23:49.294152 Step: Generate a run id if not passed from the engine/experiment 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:36
details => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2 | cut -c -6", "delta": "0:00:00.289290", "end": "2020-03-30 18:23:49.277198", "rc": 0, "start": "2020-03-30 18:23:48.987908", "stderr": "", "stderr_lines": [], "stdout": "luFnER", "stdout_lines": ["luFnER"]}

2020-03-30 18:23:49.343520 Step: set_fact 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:40
details => {"ansible_facts": {"run_id": "lufner"}, "changed": false}

2020-03-30 18:23:49.914479 Step: Obtain Container ID of the target container 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:59
details => {"changed": true, "cmd": "kubectl get pods nginx-6546cd7545-jsrzk -n default -o jsonpath='{.status.containerStatuses[?(@.name==\"nginx\")].containerID}' | awk -F 'docker://' '{print $2}'", "delta": "0:00:00.369209", "end": "2020-03-30 18:23:49.897857", "rc": 0, "start": "2020-03-30 18:23:49.528648", "stderr": "", "stderr_lines": [], "stdout": "8fd6dc3aaa54c2a7fc309aafdcaa47e16e7de850d42dd6efbb323204c3154b61", "stdout_lines": ["8fd6dc3aaa54c2a7fc309aafdcaa47e16e7de850d42dd6efbb323204c3154b61"]}

2020-03-30 18:23:49.956904 Step: Record the app_container_id 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:68
details => {"ansible_facts": {"c_container_id": "8fd6dc3aaa54c2a7fc309aafdcaa47e16e7de850d42dd6efbb323204c3154b61"}, "changed": false}

2020-03-30 18:23:50.194214 Step: Patch the chaos params to cpu stress job template 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:76
details => {"changed": true, "checksum": "fc31aad9f24b4fd1ad3b6f8f679b3f657aed3ba3", "dest": "/chaoslib/litmus/cpu_hog/app_cpu_stress.yml", "gid": 0, "group": "root", "md5sum": "5a67932584daaf6d8d1e129459f00095", "mode": "0644", "owner": "root", "size": 1061, "src": "/root/.ansible/tmp/ansible-tmp-1585592630.03-108358515051069/source", "state": "file", "uid": 0}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-30 18:23:50.663003 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.296788", "end": "2020-03-30 18:23:50.647470", "rc": 0, "start": "2020-03-30 18:23:50.350682", "stderr": "", "stderr_lines": [], "stdout": "2020-03-30T18:23:50+00:00", "stdout_lines": ["2020-03-30T18:23:50+00:00"]}

2020-03-30 18:23:50.708033 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-30T18:23:50Z"}, "changed": false}

2020-03-30 18:23:51.223400 Step: [Event]: Creating a ChaosInject event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-30T18:23:50Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "8d079fe8-72b3-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-30T18:23:50Z", "message": "Injecting pod-cpu-hog chaos on nginx-6546cd7545-jsrzk pod", "metadata": {"creationTimestamp": "2020-03-30T17:15:11Z", "name": "event-\"ChaosInject\"", "namespace": "shubham", "resourceVersion": "2077", "selfLink": "/api/v1/namespaces/shubham/events/event-%22ChaosInject%22", "uid": "038f2d84-72aa-11ea-91b9-42010a800049"}, "reason": "ChaosInject", "reportingComponent": "", "reportingInstance": "", "source": {"component": "pod-cpu-hog-sy5iws-k4p9t"}, "type": "Normal"}}
included: /utils/common/create_chaos_pod.yml for 127.0.0.1 => (item=1)
included: /utils/common/create_chaos_pod.yml for 127.0.0.1 => (item=2)
included: /utils/common/create_chaos_pod.yml for 127.0.0.1 => (item=3)

2020-03-30 18:23:51.959148 Step: Creating the chaos pod in default namespace 
task path: /utils/common/create_chaos_pod.yml:3
details => {"changed": true, "cmd": "kubectl apply -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n default", "delta": "0:00:00.514482", "end": "2020-03-30 18:23:51.941033", "rc": 0, "start": "2020-03-30 18:23:51.426551", "stderr": "", "stderr_lines": [], "stdout": "pod/app-cpu-stress-lufner created", "stdout_lines": ["pod/app-cpu-stress-lufner created"]}
FAILED - RETRYING: Check for running status of the chaos pod (90 retries left).
FAILED - RETRYING: Check for running status of the chaos pod (89 retries left).
FAILED - RETRYING: Check for running status of the chaos pod (88 retries left).
FAILED - RETRYING: Check for running status of the chaos pod (87 retries left).

2020-03-30 18:24:02.967467 Step: Check for running status of the chaos pod 
task path: /utils/common/create_chaos_pod.yml:10
details => {"attempts": 5, "changed": true, "cmd": "kubectl get pods -l name=app-cpu-stress-lufner -n default -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.354133", "end": "2020-03-30 18:24:02.947091", "rc": 0, "start": "2020-03-30 18:24:02.592958", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-30 18:24:03.071696 Step: Set pod_running_status to Running 
task path: /utils/common/create_chaos_pod.yml:41
details => {"ansible_facts": {"pod_running_status": "Running"}, "changed": false}

2020-03-30 18:24:03.405212 Step: Calculate the total wait time for cpu stress job 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:106
details => {"ansible_facts": {"job_wait_time": "180"}, "changed": false}
FAILED - RETRYING: Wait until the cpu stress job is completed (180 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (179 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (178 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (177 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (176 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (175 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (174 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (173 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (172 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (171 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (170 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (169 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (168 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (167 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (166 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (165 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (164 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (163 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (162 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (161 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (160 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (159 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (158 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (157 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (156 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (155 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (154 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (153 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (152 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (151 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (150 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (149 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (148 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (147 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (146 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (145 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (144 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (143 retries left).

2020-03-30 18:25:07.211659 Step: Wait until the cpu stress job is completed 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:111
details => {"attempts": 39, "changed": true, "cmd": "kubectl get pods -l name=app-cpu-stress-lufner --no-headers -n default --no-headers -o custom-columns=:status.phase", "delta": "0:00:00.563092", "end": "2020-03-30 18:25:07.182885", "rc": 0, "start": "2020-03-30 18:25:06.619793", "stderr": "", "stderr_lines": [], "stdout": "Succeeded", "stdout_lines": ["Succeeded"]}

2020-03-30 18:25:08.135084 Step: Tear down cpu chaos infrastructure 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:128
details => {"changed": true, "cmd": "kubectl delete -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n default", "delta": "0:00:00.681625", "end": "2020-03-30 18:25:08.114598", "rc": 0, "start": "2020-03-30 18:25:07.432973", "stderr": "", "stderr_lines": [], "stdout": "pod \"app-cpu-stress-lufner\" deleted", "stdout_lines": ["pod \"app-cpu-stress-lufner\" deleted"]}

2020-03-30 18:25:08.860382 Step: Confirm that the cpu stress pod is deleted successfully 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:134
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -l name=app-cpu-stress-lufner --no-headers -n default", "delta": "0:00:00.516933", "end": "2020-03-30 18:25:08.832127", "rc": 0, "start": "2020-03-30 18:25:08.315194", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-03-30 18:25:09.690213 Step: [Status]: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n default -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.554098", "end": "2020-03-30 18:25:09.662200", "rc": 0, "start": "2020-03-30 18:25:09.108102", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-03-30 18:25:10.379919 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n default -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.498299", "end": "2020-03-30 18:25:10.353242", "rc": 0, "start": "2020-03-30 18:25:09.854943", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-30 18:25:11.024393 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.373128", "end": "2020-03-30 18:25:11.003490", "rc": 0, "start": "2020-03-30 18:25:10.630362", "stderr": "", "stderr_lines": [], "stdout": "2020-03-30T18:25:11+00:00", "stdout_lines": ["2020-03-30T18:25:11+00:00"]}

2020-03-30 18:25:11.095674 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-30T18:25:11Z"}, "changed": false}

2020-03-30 18:25:11.939761 Step: [Event]: Creating a PostChaosCheck event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-30T18:25:11Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "8d079fe8-72b3-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-30T18:25:11Z", "message": "AUT is Running successfully", "metadata": {"creationTimestamp": "2020-03-30T17:28:28Z", "name": "event-\"PostChaosCheck\"", "namespace": "shubham", "resourceVersion": "2084", "selfLink": "/api/v1/namespaces/shubham/events/event-%22PostChaosCheck%22", "uid": "de8310a1-72ab-11ea-91b9-42010a800049"}, "reason": "PostChaosCheck", "reportingComponent": "", "reportingInstance": "", "source": {"component": "pod-cpu-hog-sy5iws-k4p9t"}, "type": "Normal"}}

2020-03-30 18:25:12.004424 Step: set_fact 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:85
details => {"ansible_facts": {"flag": "Pass"}, "changed": false}
included: /utils/runtime/getting_failure_step.yml for 127.0.0.1

2020-03-30 18:25:12.455257 Step: [Failure-Detection]: Recording the name of failed step 
task path: /utils/runtime/getting_failure_step.yml:44
details => {"ansible_facts": {"failStep": "N/A"}, "changed": false}
included: /utils/common/generate-kubernetes-chaos-events.yml for 127.0.0.1

2020-03-30 18:25:13.088473 Step: [Event]: Obtain the current date in ISO8601 format 
task path: /utils/common/generate-kubernetes-chaos-events.yml:1
details => {"changed": true, "cmd": "date --iso-8601=seconds", "delta": "0:00:00.355721", "end": "2020-03-30 18:25:13.066794", "rc": 0, "start": "2020-03-30 18:25:12.711073", "stderr": "", "stderr_lines": [], "stdout": "2020-03-30T18:25:13+00:00", "stdout_lines": ["2020-03-30T18:25:13+00:00"]}

2020-03-30 18:25:13.167329 Step: set_fact 
task path: /utils/common/generate-kubernetes-chaos-events.yml:5
details => {"ansible_facts": {"ts": "2020-03-30T18:25:13Z"}, "changed": false}

2020-03-30 18:25:13.975209 Step: [Event]: Creating a Summary event in the chaosengine 
task path: /utils/common/generate-kubernetes-chaos-events.yml:8
details => {"changed": true, "method": "patch", "result": {"apiVersion": "v1", "count": 1, "eventTime": null, "firstTimestamp": "2020-03-30T18:25:13Z", "involvedObject": {"apiVersion": "litmuschaos.io/v1alpha1", "kind": "ChaosEngine", "name": "engine", "namespace": "shubham", "uid": "8d079fe8-72b3-11ea-91b9-42010a800049"}, "kind": "Event", "lastTimestamp": "2020-03-30T18:25:13Z", "message": "pod-cpu-hog Experiment Passed!", "metadata": {"creationTimestamp": "2020-03-30T17:24:24Z", "name": "event-\"Summary\"", "namespace": "shubham", "resourceVersion": "2085", "selfLink": "/api/v1/namespaces/shubham/events/event-%22Summary%22", "uid": "4ce8dbb9-72ab-11ea-91b9-42010a800049"}, "reason": "Summary", "reportingComponent": "", "reportingInstance": "", "source": {"component": "pod-cpu-hog-sy5iws-k4p9t"}, "type": "Normal"}}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-03-30 18:25:14.840766 Step: [Result]: Update the chaos result CR to reflect EOT (End of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:23
details => {"changed": true, "checksum": "e86b8f69d5d1ee8f1e649b14bbe3cdbea15a498b", "dest": "/utils/runtime/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "8aef6e86a4e6efbfcdd402caa92ed770", "mode": "0644", "owner": "root", "size": 323, "src": "/root/.ansible/tmp/ansible-tmp-1585592714.2-241244512830483/source", "state": "file", "uid": 0}

2020-03-30 18:25:15.719825 Step: [The END]: Apply the chaos result CR for pod-cpu-hog experiment 
task path: /utils/runtime/update_chaos_result_resource.yml:33
details => {"changed": true, "cmd": "kubectl apply -f /utils/runtime/chaos-result.yml -n shubham", "delta": "0:00:00.682704", "end": "2020-03-30 18:25:15.690508", "failed_when_result": false, "rc": 0, "start": "2020-03-30 18:25:15.007804", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/engine-pod-cpu-hog configured", "stdout_lines": ["chaosresult.litmuschaos.io/engine-pod-cpu-hog configured"]}
META: ran handlers
META: ran handlers

2020-03-30 18:25:15.726963 ************ RELAX, EXPERIMENT ENDS! ************ 
127.0.0.1                  : ok=55   changed=27   unreachable=0    failed=0   


```

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml [hunk 1]
```diff
@@ -97,21 +89,28 @@
...
       when: "c_engine is defined and c_engine != ''"
 
-    - name: Setup cpu chaos infrastructure
-      shell: >
-        kubectl create -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n {{ app_ns }}
-      args: 
-        executable: /bin/bash
-      register: cpu_app_deploy_result
+      # Pod Cpu Hog pods creation is attempted for a total of 3 times, if it is not immediately schedulable due to transient node conditions
+      # If the pod-cpu-hog pod is not schedulable across these 3 tries, the experiment is failed with message indicating improper cluster state.
```
Comment (uditgaurav) [inline chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: Let's use CPU (in caps) in all places

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml [hunk 2]
```diff
@@ -97,21 +89,28 @@
...
+        pod_ns: "{{ app_ns }}"
+        c_path: "/chaoslib/litmus/cpu_hog/app_cpu_stress.yml"
+        pod_label: "name=app-cpu-stress-{{ run_id }}"
+      with_sequence: start=1 end=3
 
+    # Failing the execution, If pod-cpu-hog pod won't come to running state after three retries.
+    - fail:
+        msg: "pod_cpu_hog lib failed, Unable to create as pod-cpu-hog pod couldn't be scheduled on the {{ app_node }} node"
+      when: "pod_running_status is not defined or pod_running_status != 'Running'"
+
```
Comment (uditgaurav) [inline chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: can you put comment in line 106 above name?

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml [hunk 3]
```diff
@@ -149,19 +152,19 @@
             kubectl delete -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n {{ app_ns }} 
           args:
             executable: /bin/bash
-          when: cpu_app_deploy_result.rc == 0
+          when: "chaos_pod_result.rc == 0 "
```
Comment (uditgaurav) [inline chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: `0"`

**Pull request merged**